include/zephyr/bluetooth/audio/audio.h

```diff
@@ -1872,6 +1872,14 @@ int bt_audio_stream_disable(struct bt_audio_stream *stream);
  *  This procedure is used by a unicast client or unicast server to make a
  *  stream start streaming.
  *
+ *  For the unicast client, this will connect the CIS for the stream before
+ *  sending the start command.
+ *
+ *  For the unicast server, this will put the stream into the streaming state if
+ *  the CIS is connected (initialized by the unicast client). If the CIS is not
+ *  connected yet, the stream will go into the streaming state as soon as the
+ *  CIS is connected.
+ *
```

Comment:
I'm not sure we need go into implementation details here. What do you think?

Reply:
Hmm, maybe not, since the ISO state is not known to the caller anyhow. 

What I wanted to capture here is that for the unicast client, you can assume that the stream will go into streaming state shortly after (as we are initiating the CIS connection), but for the unicast server, the state change to streaming will depend on when the remote side connects the CIS. 

Since the caller does not know the CIS state, it cannot possible know if this will put the stream into the streaming state immediately (if CIS is connected), or later (if CIS is not connected).